### PR TITLE
Feature/#27 Movie Details integration

### PR DESCRIPTION
**Pull request by revodark**
aCapture

### CODE REVIEW

File: frontend/src/routes/AppRoutes.js [hunk 1]
```diff
@@ -22,6 +23,15 @@ const AppRoutes = () => (
       itemsArray={navBarItems}
     />
     <Switch>
+      <Route
```
Comment (GigiEillen) [inline frontend/src/routes/AppRoutes.js]: For future cases, as you are not using the render props, does it make sense to use the children directly?
```
<Route  path='/movies/now_playing'>
   <MovieCategory
          category='now_playing'
          useAxiosHook={useAxiosfourHours}
    />
</Route>

```

File: frontend/src/routes/AppRoutes.js [hunk 2]
```diff
@@ -40,6 +50,12 @@ const AppRoutes = () => (
           <MovieCategory category='upcoming' useAxiosHook={useAxiosOneHour} />
         )}
       />
+      <Route
+        path='/movies/:id'
+        render={({ match }) => (
```
Comment (GigiEillen) [inline frontend/src/routes/AppRoutes.js]: Whenever `match` is undefined won't cause a crash in the app? is it safer to use `get("params.id", match)` ?  Let me know what do you think

File: frontend/src/views/MovieCategory/MovieCategory.test.js [hunk 3]
```diff
@@ -22,7 +24,9 @@ describe('Test MovieCategory View', () => {
 
   test('renders MovieCategory view without crashes', () => {
     const { getByTestId } = render(
-      <MovieCategory category='upcoming' useAxiosHook={useAxiosHook} />
+      <BrowserRouter>
+        <MovieCategory category='upcoming' useAxiosHook={useAxiosHook} />
```
Comment (GigiEillen) [inline frontend/src/views/MovieCategory/MovieCategory.test.js]: I know this is the test, but do we have these categories as constants in any file? I'm saying this, so whenever we want to add a new type is easy to see the available ones and at some point is more organized. 
Usually, we have this kind of information from the service (own backend) so we have different configurations according what we are providing to the user.

File: backend/src/controllers/movies.test.js [hunk 4]
```diff
@@ -1,26 +1,111 @@
...
+      backdrop_path: '/2WgieNR1tGHlpJUsolbVzbUbE1O.jpg',
+      poster_path: '/iZf0KyrE25z1sage4SYFLCCrMi9.jpg',
+      release_date: '1994-09-23',
+      vote_average: 8.7,
+      vote_count: 14944,
+      genre_ids: [18, 80]
     }
   ];
+
+  const movie = {
```
Comment (Rodri77) [inline backend/src/controllers/movies.test.js]: I think we should create a file like testUtils or testMocks and there we can have these constants like movies, cast, crew, etc. We are repeating the same values on several test files and we can have that in one place.

File: backend/src/modules/utils/index.js [hunk 5]
```diff
@@ -1,3 +1,3 @@
-import { getGenreNames, mapMovieData } from './utils';
+import { getGenreNames, mapMovieData, mapMovieDetails } from './utils';
```
Comment (Rodri77) [inline backend/src/modules/utils/index.js]: Not sure if we need an index file for these, looks like we can just have utils.js and do export const from there.

File: frontend/src/routes/AppRoutes.js (hunk 2 above)
Comment (Rodri77) [reply frontend/src/routes/AppRoutes.js]: I think here we can use useParams hook from react-router-dom instead right?

File: backend/src/controllers/movies.test.js [hunk 6]
```diff
@@ -29,14 +114,37 @@ describe('Test TheMovieDB Controller', () => {
...
+      params: {
+        id: 278
+      }
+    };
+
+    const res = mockResponse();
+    getMovieDetails.mockImplementationOnce(() => Promise.resolve(movie));
+
+    await getDetails(req, res, next);
+    expect(res.status).toHaveBeenCalledWith(200);
```
[the remark this answers is not in the document]
Comment (revodark) [reply backend/src/controllers/movies.test.js]: We're using jest on the backend :) in this case we're using the [Async/Await](https://jestjs.io/docs/en/asynchronous#asyncawait) variant for testing asynchronous code, I double checked that and the tests are working, but thanks for share this other way to test, would you like us to change the return promise approach?

File: frontend/src/routes/AppRoutes.js (hunk 1 above)
Comment (revodark) [reply frontend/src/routes/AppRoutes.js]: it's great! thanks!!

File: frontend/src/routes/AppRoutes.js (hunk 2 above)
Comment (revodark) [reply frontend/src/routes/AppRoutes.js]: now I'm using useParams and  works good, thanks!!

File: backend/src/controllers/movies.test.js (hunk 4 above)
Comment (revodark) [reply backend/src/controllers/movies.test.js]: thanks it's solved

File: backend/src/modules/utils/index.js (hunk 5 above)
Comment (revodark) [reply backend/src/modules/utils/index.js]: it's solved

File: frontend/src/views/MovieCategory/MovieCategory.test.js (hunk 3 above)
Comment (revodark) [reply frontend/src/views/MovieCategory/MovieCategory.test.js]: it's solved

File: backend/src/controllers/movies.test.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (GigiEillen) [reply backend/src/controllers/movies.test.js]: There is no need to change, just to let you know is a possibility whenever we use promises.

**Pull request merged**